lib/pure/os.nim

```diff
@@ -84,6 +84,8 @@ type
 
 include "includes/osseps"
 
+proc absolutePathInternal(path: string): string {.gcsafe.}
```

Comment:
Why is this needed? And can we not forward declare it? forward decls are bad for the effect system.

Reply:
2 other options:
A1: something like https://github.com/timotheecour/Nim/pull/38 where I did the minimal changes to avoid fwd declare
it really sucks as it moves around a bunch of procs, breaks git blame/git diff history, and isn't really helping in case future PR need a dependency the other way around.

A2: https://github.com/timotheecour/Nim/pull/39 which uses `{.experimental: "codeReordering".}` 
why not? maybe it's time to use it ...

(A3: improve effect system so that it auto-infers attributes in case by solving a topological sort (analog to that in codeReordering) or fail there are cycles; maybe this would just reduce to `{.experimental: "codeReordering".}` , not sure.)

my favorite is the simple A2, or current PR, but A1 seems just ... sad

Reply:
But why is it needed? Previously we managed to do without.

Reply:
absolutePathInternal is now used in `relativePath` as part of fix for #13222:
relativePath(rel, abs) and relativePath(abs, rel) now work